Wikipedia/Code/PlacesViewController.swift

```diff
@@ -378,6 +385,55 @@ class PlacesViewController: ArticleLocationCollectionViewController, UISearchBar
         }
     }
 
+    // MARK: Profile Button
+
+    private var _yirCoordinator: YearInReviewCoordinator?
+    var yirCoordinator: YearInReviewCoordinator? {
+
+        guard let navigationController,
+              let yirDataController else {
+            return nil
+        }
+
+        guard let existingYirCoordinator = _yirCoordinator else {
+            _yirCoordinator = YearInReviewCoordinator(navigationController: navigationController, theme: theme, dataStore: dataStore, dataController: yirDataController)
+            _yirCoordinator?.badgeDelegate = self
+            return _yirCoordinator
+        }
+
+        return existingYirCoordinator
+    }
+
+    private var _profileCoordinator: ProfileCoordinator?
+    private var profileCoordinator: ProfileCoordinator? {
+
+        guard let navigationController,
+              let yirCoordinator = self.yirCoordinator else {
+            return nil
+        }
+
+        guard let existingProfileCoordinator = _profileCoordinator else {
+            _profileCoordinator = ProfileCoordinator(navigationController: navigationController, theme: theme, dataStore: dataStore, donateSouce: .savedProfile, logoutDelegate: self, sourcePage: ProfileCoordinatorSource.saved, yirCoordinator: yirCoordinator)
```

Comment:
Something interesting happens when the system automatically changes to dark or light mode and the app theme is Default.

1. From Places, open Profile and then Settings or Notifications
2. Tap back to Places.
3. Background app, change system light/dark mode
4. Go back to app, open profile, then Settings or Notifications
5. Settings or Notifications is the wrong theme

This doesn't happen with Explore, so maybe there's some sort of theme update there that isn't happening here. The Saved tab profile probably has the same bug.

Reply:
It was happening when manually changing the theme as well. I'm updating the theme we send to the profile coordinator; it seemed to fix both situations. Let me know if you catch any other inconsistencies.